### PR TITLE
Fix Misc Getter issues

### LINKED ISSUE #62: get_mac_address_table not returning expected results
Different IOS platforms format `show mac address-table` differently (different order and/or different number of fields), which can cause ValueError exceptions or in some cases empty result sets depending on the format.

Some examples:

#  6500

## Error:
```
In [26]: device.get_mac_address_table()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-26-2c330fa5b90b> in <module>()
----> 1 mac_table = device.get_mac_address_table()

/opt/napalm/env/local/lib/python2.7/site-packages/napalm_ios/ios.pyc in get_mac_address_table(self)
   1274                     'mac': mac,
   1275                     'interface': interface,
-> 1276                     'vlan': int(vlan),
   1277                     'static': static,
   1278                     'active': active,

ValueError: invalid literal for int() with base 10: 'not'
```

##  Table format:
```
Legend: * - primary entry
        age - seconds since last seen
        n/a - not available

  vlan   mac address     type    learn     age              ports
------+----------------+--------+-----+----------+--------------------------
*  999  1111.2222.3333   dynamic  Yes          0   Port-channel1
```

#  4948

## Error
```
In [41]: device.get_mac_address_table()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-41-dcfc80b6b664> in <module>()
----> 1 device.get_mac_address_table()

/opt/napalm/env/local/lib/python2.7/site-packages/napalm_ios/ios.pyc in get_mac_address_table(self)
   1282                 mac_address_table.append(entry)
   1283             else:
-> 1284                 raise ValueError("Unexpected output from: {}".format(line.split()))
   1285         return mac_address_table
   1286 

ValueError: Unexpected output from: [u'-------+---------------+--------+---------------------+--------------------']
```

##  Table format
```
Unicast Entries
 vlan   mac address     type        protocols               port
-------+---------------+--------+---------------------+--------------------
 999    1111.2222.3333   dynamic ip                    Port-channel1      
```

#  2960

##  Error
None, but result set is empty

##  Table format
```
Mac Address Table
-------------------------------------------

Vlan    Mac Address       Type        Ports
----    -----------       --------    -----
All    1111.2222.3333    STATIC      CPU
```

### LINKED ISSUE #74: Multiple Getters Failing on IOS-XE (ASR1001)
### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
- [x] Yes
- [ ] No

### napalm-ios version
(Paste verbatim output from `pip freeze | grep napalm-ios` between quotes below)

```
napalm-ios==0.4.0
napalm-iosxr==0.4.0
```

### IOS version
(Paste verbatim output from `show version` between quotes below)

```
Cisco IOS XE Software, Version 03.16.01a.S - Extended Support Release
Cisco IOS Software, ASR1000 Software (X86_64_LINUX_IOSD-UNIVERSALK9-M), Version 15.5(3)S1a, 
ROM: IOS-XE ROMMON
iosxe-rtr uptime is 5 weeks, 6 days, 17 hours, 17 minutes
Uptime for this control processor is 5 weeks, 6 days, 17 hours, 18 minutes
System returned to ROM by reload at 04:39:13 UTC Wed Oct 19 2016
System image file is "bootflash:asr1001x-universalk9.03.16.01a.S.155-3.S1a-ext.SPA.bi"
Last reload reason: Reload Command
cisco ASR1001-X (1NG) processor (revision 1NG) with 3729813K/6147K bytes of memory.
Configuration register is 0x2102
```

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
>>> device.get_bgp_neighbors()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/napalm_ios/ios.py", line 917, in get_bgp_neighbors
    out_queue, up_time, state_prefix = fields
ValueError: need more than 1 value to unpack

>>> device.get_facts()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/napalm_ios/ios.py", line 583, in get_facts
    interface = line.split()[0]
IndexError: list index out of range

>>> device.get_arp_table()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/napalm_ios/ios.py", line 1177, in get_arp_table
    raise ValueError("Unexpected output from: {}".format(line.split()))
ValueError: Unexpected output from: [u'Time', u'source', u'is', u'NTP,', u'*22:05:56.858', u'UTC', u'Tue', u'Nov', u'29', u'2016']

>>> device.get_interfaces()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/napalm_ios/ios.py", line 663, in get_interfaces
    raise ValueError(u"Unexpected Response from the device")
ValueError: Unexpected Response from the device

>>> device.get_interfaces_ip()                     
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python2.7/site-packages/napalm_ios/ios.py", line 746, in get_interfaces_ip
    raise ValueError("Unexpected response from the router")
ValueError: Unexpected response from the router
```

The BGP neighbors error is most likely due to the 'Neighbors' column in IOS not being wide enough to accomadate an IP address with four full octets (*192.168.123.123). For the get facts I'm not sure what's causing the error.

BGP neighbors example:

```
*192.168.123.97 4        12345   20902   20895    13061    0    0 3d09h           1
*192.168.123.98 4        12345   967     971    13061    0    0 03:45:19        2
*192.168.123.99 4        12345   18277   18271    13061    0    0 2d23h           2
*192.168.123.100
                4        12345    8783    8788    13061    0    0 1d10h           1
*192.168.123.101
                4        12345   16235   16202    13061    0    0 2d15h           2
*192.168.123.102
                4        12345   21769   21796    13061    0    0 3d12h           2
*192.168.123.103
                4        12345    2284    2293    13061    0    0 08:53:20        2
```

### LINKED ISSUE #76: get_arp_table() bug with static arp entries
### Description of Issue/Question

While working with the pytest framework I noticed a bug. Will just ignore this for now and remove the static entry.

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
- [x] Yes
- [ ] No

### napalm-ios version

```
napalm-ios==0.4.0
```

### IOS version
(Paste verbatim output from `show version` between quotes below)

```
Cisco IOS Software, C2960 Software (C2960-LANBASEK9-M), Version 15.0(2)SE4, RELEASE SOFTWARE (fc1)
```

### Steps to Reproduce the Issue

```
conf term
 arp 10.0.0.1 0010.2345.1cda ARPA

```

The getter is running:

```
NS2903-ASW-01#show arp | exclude Incomplete
Protocol  Address          Age (min)  Hardware Addr   Type   Interface
Internet  10.0.0.1                -   0010.2345.1cda  ARPA
Internet  172.29.50.1             6   84b8.0276.ac0e  ARPA   Vlan20
Internet  172.29.50.2           219   0019.0725.344a  ARPA   Vlan20
Internet  172.29.50.3             -   0024.f7dd.7741  ARPA   Vlan20
Internet  172.29.50.10           34   6805.ca12.71c2  ARPA   Vlan20
Internet  172.29.52.33           59   84b8.0276.ac0e  ARPA   Vlan41
Internet  172.29.52.34            -   0024.f7dd.7743  ARPA   Vlan41
Internet  172.29.52.40            1   a099.9b1c.dfa7  ARPA   Vlan41
Internet  192.168.81.34           -   0024.f7dd.7743  ARPA   Vlan41

NS2903-ASW-01#
```

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
>>> device.open()
>>> a = device.get_arp_table()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/patrick/.virtualenvs/napalm/lib/python2.7/site-packages/napalm_ios/ios.py", line 1177, in get_arp_table
    raise ValueError("Unexpected output from: {}".format(line.split()))
ValueError: Unexpected output from: [u'Internet', u'10.0.0.1', u'-', u'0010.2345.1cda', u'ARPA']
>>>
```

### LINKED ISSUE #77: Issue with get_mac_address_table() on some devices
### Description of Issue/Question

The function uses `show mac-address-table` to get mac addresses. For some IOS devices the `show mac address-table` command is used.

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
- [x] Yes
- [ ] No

### napalm-ios version

```
napalm-ios==0.4.0
```

### IOS version
(Paste verbatim output from `show version` between quotes below)

```
Cisco IOS Software, C2960 Software (C2960-LANBASEK9-M), Version 15.0(2)SE4, RELEASE SOFTWARE (fc1)
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2013 by Cisco Systems, Inc.
Compiled Wed 26-Jun-13 02:49 by prod_rel_team
```

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
>>> device.open()
>>> a = device.get_mac_address_table()
>>> a
[]
>>>

```

A solution could be to issue `show mac-address-table` and look for an error and then execute the other command instead.

```
NS2903-ASW-01#show mac-address-table
                      ^
% Invalid input detected at '^' marker.

NS2903-ASW-01#show mac address-table
          Mac Address Table
-------------------------------------------

Vlan    Mac Address       Type        Ports
----    -----------       --------    -----
 All    0100.0ccc.cccc    STATIC      CPU
 All    0100.0ccc.cccd    STATIC      CPU
 All    0180.c200.0000    STATIC      CPU
 All    0180.c200.0001    STATIC      CPU
 All    0180.c200.0002    STATIC      CPU
 All    0180.c200.0003    STATIC      CPU
<etc>
```

### PR DESCRIPTION
**Pull request by ktbyers**
Fix #74 BGP getters issue with trailing newline and additional terminal messages in output.
Fix #77 to support multiple show mac address commands
Fix #76 support for static ARP entries
Fix #62 support multiple formats for show mac address output

### CODE REVIEW

**Comment by ktbyers** [comment]
I also added a wrapper that would strip out the following lines from the output:

```
Load for five secs: 1%/0%; one minute: 1%; five minutes: 1%
Time source is NTP, *09:59:40.324 UTC Wed Nov 30 2016
```

**Comment by dbarrosop** [comment]
Could you add this to `setup.cfg`, please?

```
[tool:pytest]
addopts = --cov=./ -vs
json_report = report.json
jsonapi = true
```

I want to be able to see on travis that the old and new test cases are running.

**Comment by ktbyers** [comment]
@dbarrosop Okay, added your fix.

**Comment by ktbyers** [comment]
Fixes issue #77 also.

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -30,6 +30,14 @@
 WEEK_SECONDS = 7 * DAY_SECONDS
 YEAR_SECONDS = 365 * DAY_SECONDS
 
+# IP Address
+IP_ADDR_REGEX = r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}"
+RE_IPADDR_STRIP = re.compile(r"({})\n".format(IP_ADDR_REGEX))
+
+IOS_COMMANDS = {
+   'show_mac_address': ['show mac-address table', 'show mac address-table'],
```
Comment (ogenstad) [inline napalm_ios/ios.py]: The first command is actually wrong here and it wouldn't get picked up by the tests. It should be `show mac-address-table`.

Comment (ktbyers) [reply napalm_ios/ios.py]: Okay...will fix.

**Comment by ktbyers** [comment]
@ogenstad Okay, command error is fixed.

**Comment by ktbyers** [comment]
@mirceaulinic @dbarrosop Note, for static ARP entry I have it return a null string for the interface. Is this what we want the behavior to be?

See issue #76

**Comment by dbarrosop** [comment]
+100

**Pull request merged**